analytics-webapps/src/main/webapp/vue-app/generic-portlet/chart/SelectPeriod.vue

```diff
@@ -6,15 +6,15 @@
       <v-btn
         class="btn"
         v-on="on">
-        {{ selectedItem && selectedItem.text }}
+        {{ selectedItem && $t(`analytics.periodOptions.`.concat(selectedItem.value)) }}
```

Comment:
It's better to never use concat in Javascript
```suggestion
        {{ selectedItem && $t(`analytics.periodOptions.${selectedItem.value}`) }}
```

BTW, you can simply change `periodOptions` this way:
computed: {
      periodOptions() {
        return [
        {
          text: this.$t('analytics.periodOptions.last24h'),
          value: 'last24h'
        },{
          text: this.$t('analytics.periodOptions.lastWeek'),
          value: 'lastWeek'
        },{
          text: this.$t('analytics.periodOptions.lastMonth'),
          value: 'lastMonth'
        },{
          text: this.$t('analytics.periodOptions.last3Months'),
          value: 'last3Months'
        },{
          text: this.$t('analytics.periodOptions.last6Months'),
          value: 'last6Months'
        },{
          text: this.$t('analytics.periodOptions.lastYear'),
          value: 'lastYear'
        },
      ];
    }
}

Reply:
done

Reply:
Note that `periodOptions` is added in `computed` section because sometimes, when adding and operation in `data` section, an NPE is thrown